clients/kafka-common/src/main/java/org/eclipse/hono/client/kafka/consumer/AsyncHandlingAutoCommitKafkaConsumer.java

```diff
@@ -0,0 +1,448 @@
+     *                      value marks the record handling as finished so that the corresponding offset may be committed.
+     * @param consumerConfig The Kafka consumer configuration.
+     * @throws NullPointerException if any of the parameters is {@code null}.
+     * @throws IllegalArgumentException if the consumerConfig is missing a "group.id" entry.
+     */
+    public AsyncHandlingAutoCommitKafkaConsumer(
+            final Vertx vertx,
+            final Pattern topicPattern,
+            final Function<KafkaConsumerRecord<String, Buffer>, Future<Void>> recordHandler,
+            final Map<String, String> consumerConfig) {
+        this(new AtomicReference<>(), vertx, null, topicPattern, recordHandler, consumerConfig);
+    }
+
+    private AsyncHandlingAutoCommitKafkaConsumer(
+            final AtomicReference<AsyncHandlingAutoCommitKafkaConsumer> selfRef,
+            final Vertx vertx,
+            final Set<String> topics,
+            final Pattern topicPattern,
+            final Function<KafkaConsumerRecord<String, Buffer>, Future<Void>> recordHandler,
+            final Map<String, String> consumerConfig) {
+        super(vertx, topics, topicPattern, mapRecordHandler(selfRef, recordHandler), validateAndAdaptConsumerConfig(consumerConfig));
+        selfRef.setPlain(this);
+
+        this.commitIntervalMillis = getCommitInterval(consumerConfig);
+        this.maxAgeBeforeRecommitSeconds = getOffsetsRecommitSeconds(consumerConfig);
+    }
+
+    private static Handler<KafkaConsumerRecord<String, Buffer>> mapRecordHandler(
+            final AtomicReference<AsyncHandlingAutoCommitKafkaConsumer> selfRef,
+            final Function<KafkaConsumerRecord<String, Buffer>, Future<Void>> recordHandler) {
+        return record -> {
+            final OffsetsQueueEntry offsetsQueueEntry = selfRef.getPlain().setRecordReceived(record);
+            recordHandler.apply(record)
+                    .onComplete(ar -> offsetsQueueEntry.setHandlingComplete());
+        };
+    }
+
+    private static Map<String, String> validateAndAdaptConsumerConfig(final Map<String, String> consumerConfig) {
+        if (Strings.isNullOrEmpty(consumerConfig.get(ConsumerConfig.GROUP_ID_CONFIG))) {
+            throw new IllegalArgumentException(ConsumerConfig.GROUP_ID_CONFIG + " config entry has to be set");
+        }
+        consumerConfig.put(ConsumerConfig.ENABLE_AUTO_COMMIT_CONFIG, "false");
+        return consumerConfig;
+    }
+
+    private static long getCommitInterval(final Map<String, String> consumerConfig) {
+        return Optional.ofNullable(consumerConfig.get(ConsumerConfig.AUTO_COMMIT_INTERVAL_MS_CONFIG))
+                .map(Long::parseLong)
+                .orElse(DEFAULT_COMMIT_INTERVAL.toMillis());
+    }
+
+    private static long getOffsetsRecommitSeconds(final Map<String, String> consumerConfig) {
+        return Optional.ofNullable(consumerConfig.get(CONFIG_HONO_OFFSETS_RECOMMIT_SECONDS))
```

Comment:
I would not mix this property into the map with the standard Kafka properties. They are usually filled by Spring or Quarkus based on a prefix and you would probably have to use non-obvious tricks to add custom properties into it.

Reply:
I meant this property to be set like the other Kafka properties:
````
hono:
  kafka:
    consumerConfig:
      bootstrap.servers: ${hono.kafka.bootstrap.servers}
      hono.offsets.skip.recommit.period.seconds: 600
````
It's a non-intrusive way to add such a property and it is something to be set for all consumers (using the above class) since it depends on the server `offsets.retention.minutes` config setting.
On the other hand I agree that it's not optimal to mix official and hono-defined properties on that level. I tried to at least make the distinction clear with the `hono.` prefix.
However, when thinking about an alternative place where to put such a property, there currently seems to be no particularly good place either. The `hono.command` section (see [here](https://github.com/eclipse/hono/blob/f69b2cb0d56b18a9288bb5b8e7cb5e7e95eb4c23/tests/src/test/resources/commandrouter/application.yml#L40)) contains only AMQP related properties.

I see there's a similar case with the `AbstractAtLeastOnceKafkaConsumer#pollTimeout` property, being defined in a `KafkaConsumerConfigProperties` field. But from what I see, there's currently no way to set this property in the `application.yml`.

Reply:
Yes, `AbstractAtLeastOnceKafkaConsumer#pollTimeout` is a similar case. The property is picked up when you use an `application.yml` containing this:
```
hono:
  kafka:
    pollTimeout: 200
    consumerConfig:
      bootstrap.servers: ${hono.kafka.bootstrap.servers}
```

So, the difference is that it is a separate member of the `KafkaConsumerConfigProperties` and not part of the configuration of the (native) Kafka consumer. Even though the Kafka client ignores properties that it does not know and only logs a warning,  it seems cleaner to me to keep Hono's properties and Kafka's properties separate.

Reply:
Ah ok, I only had a look at the quarkus implementation here for which this doesn't seem to be implemented yet.

In any case, I agree that we should separate the Hono and standard Kafka config properties. I think to achieve this properly, we should reconsider the property structure in general, e.g. by using a 
````
consumer:
  pollTimeout: 200
  skipOffsetRecommitPeriodSeconds: 600
  clientConfig: 
    bootstrap.servers: ${hono.kafka.bootstrap.servers}
    [...]
````
structure, possibly put e.g. under `hono.command.kafka`, to enable defining different properties for different consumer use cases.
I think this should be done in a follow-up PR then. Until then, I would prefer to keep the code here as is.